src/MDMMotionAnimator.m

```diff
@@ -100,7 +100,8 @@ - (void)animateWithTiming:(MDMMotionTiming)timing
 
       // When we use a nil key, Core Animation will ensure that the animation is added with a
       // unique key - this enables our additive animations to stack upon one another.
-      [layer addAnimation:animation forKey:nil];
+      NSString *key = _additive ? nil : keyPath;
+      [layer addAnimation:animation forKey:key];
```

Comment:
Does CoreAnimation remove the old animation and then add the new one or is the new one ignored until the old one is removed?

For example, are these equivalent?
```objectivec
[layer addAnimation:anim1 forKey:key];
[layer addAnimation:anim2 forKey:key]; // Removes anim1, then adds anim2 instead
```

```objectivec
[layer addAnimation:anim1 forKey:key];
[layer removeAnimationForKey:key];
[layer addAnimation:anim2 forKey:key];
```

Reply:
Yep.

https://developer.apple.com/documentation/quartzcore/calayer/1410848-addanimation

> A string that identifies the animation. Only one animation per unique key is added to the layer. The special key kCATransition is automatically used for transition animations. You may specify nil for this parameter.

Reply:
Yeah, I had read the documentation and noticed that it didn't explicitly mention replacing any current animations on the object. Thanks!